Add Snakefile for this step

qubixes opened the pull request:
This adds basic automation to the Carbolite workflow. It locally creates the .mat file and uploads it to YoDa (currently the wrong collection, but I do not have access to the correct one). It is probably Linux/MacOS only because of the piping to circumvent the SyntaxError. 

Fixes #2

EmiliaJarochowska commented:
It seems to work (even though I didn't do exactly [the installation as in snakemake's docs](https://snakemake.readthedocs.io/en/stable/getting_started/installation.html)) but I got the following error:
```
WorkflowError in file /home/ejarochows/Triangle_workflow/Snakefile, line 16:
No storage provider for query irods://nluu12p/home/research-test-christine/triangle/{param_file}.mat. Either install the required storage plugin or check your query. Also consider to explicitly specify the storage provider to get a more informative error message.
```
So I thought this was because I didn't have access to that Yoda folder so I changed the paths in lines 3 and 16 to my folder:
`irods://nluu11p/home/research-mindthegap/triangle` instead of `irods://nluu12p/home/research-test-christine/triangle/` but still get the same error. My connection to irods definitely works, I can do `ibridges list` and see my folders. Is perhaps `{param_file}` not parsed correctly?

qubixes commented:
> It seems to work (even though I didn't do exactly [the installation as in snakemake's docs](https://snakemake.readthedocs.io/en/stable/getting_started/installation.html)) but I got the following error:
> 
> ```
> WorkflowError in file /home/ejarochows/Triangle_workflow/Snakefile, line 16:
> No storage provider for query irods://nluu12p/home/research-test-christine/triangle/{param_file}.mat. Either install the required storage plugin or check your query. Also consider to explicitly specify the storage provider to get a more informative error message.
> ```
> 
> So I thought this was because I didn't have access to that Yoda folder so I changed the paths in lines 3 and 16 to my folder: `irods://nluu11p/home/research-mindthegap/triangle` instead of `irods://nluu12p/home/research-test-christine/triangle/` but still get the same error. My connection to irods definitely works, I can do `ibridges list` and see my folders. Is perhaps `{param_file}` not parsed correctly?

Sorry, the installation instructions were not complete. You also need the snakemake-irods/ibridges plugin for the workflow to work. I have added a command to install it. Alternatively you can clone the repo, switch to the branch and pip install it that way.

EmiliaJarochowska reviewed (approved):
Thanks, works fantastic ❤️
